pkg/gitops/client.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitops
+
+import (
+	"context"
+	"os"
+
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	networking "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type K8sClient struct {
+	client.Client
+}
+
+func NewKubernetes() (*K8sClient, error) {
+	var config *rest.Config
+	kubeconfig, hasIt := os.LookupEnv("KUBECONFIG")
+	if !hasIt {
+		kubeconfig = clientcmd.RecommendedHomeFile
+	}
+	if _, err := os.Stat(kubeconfig); kubeconfig == clientcmd.RecommendedHomeFile && os.IsNotExist(err) {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			klog.ErrorS(err, "unable to create InClusterConfig")
+			return nil, err
+		}
```

Comment:
I thought it was more common to start with `InClusterConfig` first and fallback to the env variable / default file otherwise. Is this a conscious choice here?

Reply:
No, I think I just misinterpreted what you said before